app/models/medium.rb

```diff
@@ -393,12 +393,6 @@ def self.similar_courses(search_string)
     end
   end
 
-  def restricted?
-    return false unless teachable
-
-    teachable.restricted?
-  end
```

Comment:
Can you confirm my observation that with the removal of the first `restricted?` method, the `restricted?` methods for courses and lessons are no longer needed (whereas the `restricted?` method for lectures is still in use)?

Reply:
Can confirm partially:

- `restricted?` method for lectures is still needed
- `restricted?` method for course is no longer needed (if you remove this, please also remove the respective test in `course_spec.rb`).
- I have not found a `restricted?` method in `lesson.rb`, it just delegates to the `restricted?` method of the lecture. If you mean that this method is not called on a lesson, you are right; out of the 15 occurrences of `restricted?` in the code, none was a call to the delegated `restricted?` method of lesson. However: is it somehow possible that a lesson is " in disguise" a lecture since a lecture has many lessons? If not, it'd be fine to even remove the `delegate :restricted?, to: :lecture` in `lesson.rb`.

Reply:
I removed the `restricted?` methods for course and lesson and the corresponding test in [https://github.com/MaMpf-HD/mampf/pull/566/commits/9fa990a30ccd1cb6348c1ed302060a77d5786a6c](https://github.com/MaMpf-HD/mampf/pull/566/commits/9fa990a30ccd1cb6348c1ed302060a77d5786a6c).